pkg/planner/core/constraint/exprs.go

```diff
@@ -40,3 +41,16 @@ func DeleteTrueExprs(p base.LogicalPlan, conds []expression.Expression) []expres
 	}
 	return newConds
 }
+
+// IsConstFalse is used to check whether the expression is a constant false expression.
+func IsConstFalse(expr expression.Expression) bool {
+	if e, ok := expr.(*expression.ScalarFunction); ok {
+		switch e.FuncName.L {
+		case ast.LT, ast.LE, ast.GT, ast.GE, ast.EQ, ast.NE:
+			if constExpr, ok := e.GetArgs()[1].(*expression.Constant); ok && constExpr.Value.IsNull() && constExpr.DeferredExpr == nil {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
We can operate the same check when building the scalar function or in constant folding.